docs/spec/lightclient/detection.md

```diff
@@ -0,0 +1,491 @@
+*sign-skip-match(a,b,c,t) = true* iff
+   - *sequ-rooted(a)* and
+   - *sequ-rooted(b)* and
+   - *b.Header.Height = c.Header.Height* and
+   - *skip-root(a,c,t)*
+
+implies *b = c*.
+
+If there exists three light blocks a, b, and c, with 
+*sign-skip-match(a,b,c,t) =
+false* then we have a *slashable fork*.
+
+
+
+> **TODO:** I think the following definition is
+> the intuition behind **main chain forks**
+> in the document on [forks][tendermintfork]. However, main chain
+> forks were defined more operational "forks that are observed by
+> full nodes as part of normal Tendermint consensus protocol". Please
+> confirm! 
+  
+#### **[TMBC-SIGN-UNIQUE]**
+Let *b* and *c* be  light blocks, we define *sign-unique(b,c) =
+true* iff
+   - *b.Header.Height =  c.Header.Height* and
+   - *sequ-rooted(b)* and
+   - *sequ-rooted(c)*
+   
+implies *b = c*.
+
+If there exists two light blocks b and c, with *sign-unique(b,c) =
+false* then we have a *fork on the chain*.
+
+> The following captures what I believe is called a light client fork
+> in our discussions. There is no fork on the chain up to the height 
+> of block b. However, c is of that height, is different, and passes skipping
+> verification  
+> Observe that a light client fork is a special case of a slashable
+> fork. 
+
+
+#### **[TMBC-LC-FORK]**
+Let *a*, *b*, *c*, be light blocks and *t* a time. We define
+*light-client-fork(a,b,c,t)* iff
+   - *sign-skip-match(a,b,c,t) = false* and
+   - *sequ-rooted(b)* and
+   - *b* is "unique", that is, for all *d*,  *sequ-rooted(d)* and
+     *d.Header.Height=b.Header.Height* implies *d = b*
+
+
+> Finally, let's also define bogus blocks that have no support.
+> Observe that bogus is even defined if there is a fork on the chain.
+
+#### **[TMBC-BOGUS]**
+Let *b* be a light block and *t* a time. We define *bogus(b,t)* iff
+  - *sequ-rooted(b) = false* and
+  - for all *a*, *sequ-rooted(a)* implies *skip-root(a,b,t) = false*
+
+> Relation to [fork accountability][accountability]: F1, F2, and F3
+> (equivocation, amnesia, back to the past) can lead to a fork on the
+> chain and to a light client fork.  
+> F4 and F5 (phantom validators, lunatic) cannot lead to a fork on the
+> chain but to a light client
+> fork if *t+1 < f < 2t+1*.  
+> F4 and F5 can also lead to bogus blocks
+
+
+
+### Informal Problem statement
+
+> We put tags to informal problem statements as there is no sequential
+> specification.
+
+The following requirements are operational in that they describe how
+things should be done, rather than what should be done. However, they
+do not constitute temporal logic verification conditions. For those,
+see [LCD-DIST-*] below.
+
+
+#### **[LCD-IP-STATE]**
+
+The detector works on a LightStore that contains LightBlocks in one of 
+the state `StateUnverified`, ` StateVerified`, `StateFailed`, or
+`StateTrusted`.
+
+
+#### **[LCD-IP-Q]**
+
+Whenever the light client verifier performs `VerifyToTarget` with the
```

Comment:
We should clarify what is the state of light blocks required to be used for VerifyToTarget?

Reply:
I guess, this should rather go to the part on the main logic. We need to discuss where to put it. I guess we should do a light client spec that uses the functions provided in the verification and detection specs